extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -201,6 +209,36 @@ export class SubscriptionService {
     return this.dialog_;
   }
 
+  /**
+   * Select and activate all platforms. Do nothing if the viewer can authorize the user.
+   * @return {!Promise}
+   */
+  selectAndActivatePlatform() {
```

Comment:
This method is untested, given it plays with viewer provided auth, or platform config, all these scenarios need to be extensively tested.

Reply:
also can we call this `renderDialogForSelectedPlatform` like it was before deletion, I think its more clear whats happening. Also the old comment was helpful. https://github.com/ampproject/amphtml/pull/19053/files#diff-db86b2ff1f6f6724963d244acc13acb7f4fe6c942192a947c55d495ef8fc77adR380

Reply:
Done.

Reply:
Sorry missed the comment. Changed the function name and prefix it with a 'maybe'.